src/LiveComponent/src/Attribute/BeforeReRender.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\LiveComponent\Attribute;
+
+/**
+ * Call a method before re-rendering.
+ *
+ * This hook ONLY happens when rendering via HTTP: it does
+ * not happen during the initial render of a component.
+ *
+ * @Annotation
+ * @Target("METHOD")
+ *
+ * @experimental
+ */
+final class BeforeReRender
```

Comment:
```suggestion
#[\Attribute(\Attribute::TARGET_METHOD)]
final class BeforeReRender
```

Reply:
Good catch! The LAST little bit that we didn't *quite* finish was for the PHP 8 attributes. That will come shortly (even quite possibly today). In addition to this change, there are some changes to the code that parses through the annotations/attributes :)

Reply:
Ooh indeed, there is of course more to do than adding some attributes :D